localstack/services/apigateway/invocations.py

```diff
@@ -388,11 +388,24 @@ def invoke_rest_api_integration_backend(invocation_context: ApiInvocationContext
                 response = result
             else:
                 response = LambdaResponse()
+                response.headers.update({"content-type": "application/json"})
+
                 parsed_result = (
                     result if isinstance(result, dict) else json.loads(str(result or "{}"))
                 )
                 parsed_result = common.json_safe(parsed_result)
                 parsed_result = {} if parsed_result is None else parsed_result
+
```

Comment:
This is outdated at this stage, I recently separated `AWS` from `AWS_PROXY` requests, thus this need to be worked out before merging. The structured response happens for `AWS_PROXY` requests, where there is an expected format, headers, statusCode and body